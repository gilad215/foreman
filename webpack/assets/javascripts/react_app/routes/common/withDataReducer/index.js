import Immutable from 'seamless-immutable';

const initialState = Immutable({
  data: null,
  isLoading: true,
  hasError: false,
  message: { type: 'empty', text: '' },
});

const withDataReducer = controller => (
  state = initialState,
  { type, payload }
) => {
  switch (type) {
    case `${controller}_PAGE_DATA_RESOLVED`:
      return state.merge(payload);

    case `${controller}_PAGE_DATA_FAILED`:
      return state.merge(payload);

    case `${controller}_PAGE_HIDE_LOADING`:
      return state.set('isLoading', false);

    default:
      return state;
  }
};

export default withDataReducer;
