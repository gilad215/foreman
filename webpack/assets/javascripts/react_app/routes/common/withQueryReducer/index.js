import Immutable from 'seamless-immutable';

const initialState = Immutable({
  page: 1,
  perPage: 25,
  searchQuery: '',
  itemCount: 0,
});

const withQueryReducer = controller => (
  state = initialState,
  { type, payload }
) => {
  switch (type) {
    case `${controller}_PAGE_UPDATE_QUERY`:
      return state.merge(payload);

    default:
      return state;
  }
};

export default withQueryReducer;
