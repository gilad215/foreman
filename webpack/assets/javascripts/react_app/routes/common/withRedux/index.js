import React from 'react';
import { mapSelectorsToProps, mapActionsToProps } from './helpers';

export const withRedux = (
  selectors,
  actions
) => Component => componentProps => (
  <Component
    {...Object.assign(
      mapSelectorsToProps(selectors),
      mapActionsToProps(actions),
      componentProps
    )}
  />
);
