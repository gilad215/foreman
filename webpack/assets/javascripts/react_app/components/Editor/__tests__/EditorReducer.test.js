import {
  EDITOR_INITIALIZE,
  EDITOR_CHANGE_STATE,
  EDITOR_IMPORT_FILE,
  EDITOR_REVERT_CHANGES,
  EDITOR_EXEC_PREVIEW,
  EDITOR_MODAL_TOGGLE,
} from '../EditorConstants';

import reducer from '../EditorReducer';

import { testReducerSnapshotWithFixtures } from '../../../common/testHelpers';

const fixtures = {
  'should return the initial state': {},
  'should handle EDITOR_INITIALIZE': {
    action: {
      type: EDITOR_INITIALIZE,
      payload: { initialState: { value: 'newValue' } },
    },
  },
  'should handle EDITOR_CHANGE_STATE': {
    action: {
      type: EDITOR_CHANGE_STATE,
      payload: {
        newState: { value: '</>' },
      },
    },
  },
  'should handle EDITOR_IMPORT_FILE': {
    action: {
      type: EDITOR_IMPORT_FILE,
      payload: {
        value: '</>',
      },
    },
  },
  'should handle EDITOR_REVERT_CHANGES': {
    action: {
      type: EDITOR_REVERT_CHANGES,
      payload: {
        value: '</>',
      },
    },
  },
  'should handle EDITOR_EXEC_PREVIEW': {
    action: {
      type: EDITOR_EXEC_PREVIEW,
      payload: {
        renderedValue: '</>',
      },
    },
  },
  'should handle EDITOR_MODAL_TOGGLE': {
    action: {
      type: EDITOR_MODAL_TOGGLE,
    },
  },
};

describe('Editor reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
