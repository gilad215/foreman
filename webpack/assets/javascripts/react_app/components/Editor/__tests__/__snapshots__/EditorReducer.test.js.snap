// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor reducer should handle EDITOR_CHANGE_STATE 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "hosts": Object {},
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "readOnly": false,
  "renderedValue": "",
  "selectedView": "input",
  "showError": false,
  "theme": "Monokai",
  "value": "</>",
}
`;

exports[`Editor reducer should handle EDITOR_EXEC_PREVIEW 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "hosts": Object {},
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "readOnly": false,
  "renderedValue": undefined,
  "selectedView": "input",
  "showError": false,
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_IMPORT_FILE 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "hosts": Object {},
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "readOnly": false,
  "renderedValue": "",
  "selectedView": "input",
  "showError": false,
  "theme": "Monokai",
  "value": "</>",
}
`;

exports[`Editor reducer should handle EDITOR_INITIALIZE 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "hosts": Object {},
  "initialState": Object {
    "value": "newValue",
  },
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "readOnly": false,
  "renderedValue": "",
  "selectedView": "input",
  "showError": false,
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_MODAL_TOGGLE 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "hosts": Object {},
  "isMasked": false,
  "isMaximized": true,
  "isRendering": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "readOnly": false,
  "renderedValue": "",
  "selectedView": "input",
  "showError": false,
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_REVERT_CHANGES 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "hosts": Object {},
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "readOnly": false,
  "renderedValue": "",
  "selectedView": "input",
  "showError": false,
  "theme": "Monokai",
  "value": "</>",
}
`;

exports[`Editor reducer should return the initial state 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "hosts": Object {},
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "readOnly": false,
  "renderedValue": "",
  "selectedView": "input",
  "showError": false,
  "theme": "Monokai",
  "value": "",
}
`;
